marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/entitySearch/getOpticPlanTest.sjs

```diff
@@ -0,0 +1,89 @@
+const search = require('/MarkLogic/appservices/search/search');
+const test = require("/test/test-helper.xqy");
+const op = require('/MarkLogic/optic');
+
+const queryOptions = '<options xmlns="http://marklogic.com/appservices/search"/>';
+
+function invokeOpticPlanService(schemaName, viewName, limit, structuredQuery, searchText, sortOrder, columns, retries=0) {
+    try {
+        // Need to clear the modules cache or unrelated error could be thrown. Investigating to file a bugtrack.
+        xdmp.moduleCacheClear();
+        return fn.head(xdmp.invoke(
+            "/data-hub/5/data-services/entitySearch/getOpticPlan.sjs",
+            {
+                schemaName,
+                viewName,
+                limit,
+                structuredQuery,
+                searchText,
+                queryOptions,
+                sortOrder: xdmp.toJSON(sortOrder).root,
+                columns: Sequence.from(columns)
+            }
+        ));
+    } catch (e) {
+        if (e.code === 'SQL-TABLEREINDEXING' || e.name === 'SQL-TABLEREINDEXING' && retries < 10) {
```

Comment:
Oof - this is rough for what we hope will be very fast tests (at least faster than the Java tests). How about a sleep of something smaller, like 200 or so? Shouldn't the reindexing be very fast due to their only being a few documents to index?

Reply:
I started with a lower number before and it would go over 10 retries. On my machine, the first time around it has typically been the third retry that is successful. I can up the max retries and bring the sleep time down.

Reply:
I think a large number of retries is fine - it's fine if the total wait amount is pretty long, we just want it to finish sleeping as soon as the reindexer is done.